azkaban-common/src/main/java/azkaban/executor/AlerterHolder.java

```diff
@@ -0,0 +1,193 @@
+/*
+ * Copyright 2017 LinkedIn Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+
+package azkaban.executor;
+
+import azkaban.alert.Alerter;
+import azkaban.utils.Emailer;
+import azkaban.utils.FileIOUtils;
+import azkaban.utils.Props;
+import azkaban.utils.PropsUtils;
+import com.google.inject.Inject;
+import java.io.File;
+import java.lang.reflect.Constructor;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.net.URLClassLoader;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.apache.log4j.Logger;
+
+
+public class AlerterHolder {
+  private final Map<String, Alerter> alerters;
+
+  private static Logger logger = Logger.getLogger(AlerterHolder.class);
```

Comment:
I think migrating to a different logger in an incremental way is difficult. More so because people just copy the logger line from a different class. This needs to be done either all at once or one module at a time. and that dependency needs to be removed.